htdocs/core/class/commondocgenerator.class.php

```diff
@@ -559,6 +559,14 @@ public function get_substitutionarray_object($object, $outputlangs, $array_key =
 			$resarray = $this->fill_substitutionarray_with_extrafields($object, $resarray, $extrafields, $array_key, $outputlangs);
 		}
 
+		// Load product data optional fields to the line -> enables to use "line_options_{extrafield}"
```

Comment:
$line here is a random value so not reliable (it as value of the last line defined by loop on line 519).

Reply:
Yes, I see.
Thanks for the reply. 

I finally found out that the functionality was already implemented in get_substitutionarray_lines (line 669).

You can close the pull request.